sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -4557,6 +4557,47 @@ object functions {
    */
   def hour(e: Column): Column = withExpr { Hour(e.expr) }
 
+  /**
+   * Extracts a part of the date/timestamp or interval source.
+   *
+   * @param field selects which part of the source should be extracted.
+   * @param source a date/timestamp or interval column from where `field` should be extracted.
+   * @return a part of the date/timestamp or interval source
+   * @group datetime_funcs
+   * @since 3.5.0
+   */
+  def extract(field: Column, source: Column): Column = withExpr {
+    UnresolvedFunction("extract", Seq(field, source).map(_.expr), isDistinct = false)
```

Comment:
I just notice that we directly use `UnresolvedFunction` before, e.g. `floor` / `lpad` / `ceil`

would you mind also change `endswith`, `startswith`, `contains` in this PR

Reply:
to narrow the related tickets, let me send the followup

Reply:
@zhengruifeng Sorry, I saw the comment too late.

Reply:
no worries, I send it in another pr